src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,177 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node node;
+    private Node head = null;
+    private Node tail = null;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        node = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = node;
+            size++;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+            node.next = null;
+            tail = node;
+            size++;
+        }
+
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index out of bounds");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> current = findNodeByIndex(index);
```

Comment:
Consider creating a `checkIndex` method to verify if the index is valid and use it here instead of the explicit check to avoid code duplication.